tests/api/test_flows.py

```diff
@@ -1102,6 +1102,72 @@ async def test_schedule_does_not_overwrite_flow_labels(self, project_id, labels)
         assert all([fr.labels == labels for fr in flow_runs[::2]])
         assert all([fr.labels == ["bar", "foo"] for fr in flow_runs[1::2]])
 
+    async def test_doesnt_schedule_same_time_twice(self, project_id):
+        now = pendulum.now("UTC")
+        clock1 = prefect.schedules.clocks.IntervalClock(
+            start_date=now.add(minutes=1),
+            interval=datetime.timedelta(minutes=2),
+        )
+        clock2 = prefect.schedules.clocks.IntervalClock(
+            start_date=now.add(minutes=1),
+            interval=datetime.timedelta(minutes=2),
+        )
+
+        flow = prefect.Flow(
+            name="Test Scheduled Flow",
+            schedule=prefect.schedules.Schedule(clocks=[clock1, clock2]),
+        )
+        flow.add_task(prefect.Parameter("x", default=1))
+        flow_id = await api.flows.create_flow(
+            project_id=project_id, serialized_flow=flow.serialize()
+        )
+        await models.FlowRun.where({"flow_id": {"_eq": flow_id}}).delete()
+        assert len(set((await api.flows.schedule_flow_runs(flow_id)))) == 10
+
+        flow_runs = await models.FlowRun.where({"flow_id": {"_eq": flow_id}}).get(
+            selection_set={"parameters": True, "scheduled_start_time": True},
+            order_by={"scheduled_start_time": EnumValue("asc")},
+        )
+
+        assert len(set([fr.scheduled_start_time for fr in flow_runs])) == 10
+
+    @pytest.mark.parametrize(
+        "attrs",
+        [[dict(parameter_defaults=dict(x="a")), dict(parameter_defaults=dict(x="b"))]],
```

Comment:
Ah, I need to add some varying labels as well.  Will update to parametrize over more possibilities.